src/components/settings/SpoolmanSettings.vue

```diff
@@ -221,6 +238,38 @@ export default class SpoolmanSettings extends Mixins(StateMixin) {
     })
   }
 
+  get availableFieldsToShowInSpoolmanCard () {
+    return [
+      'id',
+      'vendor',
+      'filament_name',
+      'remaining_weight',
+      'used_weight',
+      'location',
+      'material',
+      'lot_nr',
+      'price',
+      'density',
+      'extruder_temp',
+      'bed_temp',
+      'first_used',
+      'last_used',
+      'comment'
+    ].map(field => ({ value: field, text: this.$t(`app.spoolman.label.${field}`) }))
+  }
+
+  get fieldsToShowInSpoolmanCard (): string[] {
```

Comment:
Is it worth using a custom type with all fields instead of `string[]`?

Something like:

```typescript
type SpoolmanKnownFields = 'id' | 'vendor' | 'filament_name' | 'remaining_weight' | 'used_weight' | 'location' | 'material' | 'lot_nr' | 'price' | 'density' | 'extruder_temp' | 'bed_temp' | 'first_used' | 'last_used' | 'comment'
```

Reply:
IMO there's no need to add an extra type, the values are never referenced in code anyway (so no compiler warnings).. And I think in the future we could also add custom spoolman fields to the dashboard, which wouldn't be covered by the type

Reply:
They are referenced on the if's, but I am fine either way! 🙂